### ISSUE TITLE
Be able to use queries outside of the pages

### ISSUE THREAD

**Issue by oliviertassinari** [mui/mui-toolpad#1523]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

I would like to be able to use queries from one page on another one.

### Examples 🌈

In https://master--toolpad.mui.com/deploy/cl6rqzry10009arlv9sto6qja/pages, we aim to get all the tables to 0 rows. There are quite a few to open. It would be great if we could create an "overview" app.

**Comment by Janpot** [comment]
Alternatively, what you may want is to have a single page and a tab component to create your different views?
You want to share things across pages, but maybe there shouldn't be different pages to begin with?

**Comment by oliviertassinari** [comment]
@Janpot I think that tabs could be another solution, yes. As long as these tabs can have their own URLs.

**Comment by prakhargupta1** [comment]
It is solved for the pro-code queries as all functions of queries.ts are now available across all pages. 
But this issue is till relevant for HTTPS client based queries.

**Comment by oliviertassinari** [comment]
@prakhargupta1 True, I suspect that we can easily solve this with the pro-code integration now. Once we have client-based queries, we could import from a shared file.
